src/core/Entry.cpp

```diff
@@ -654,7 +651,28 @@ void Entry::setOverrideUrl(const QString& url)
 
 void Entry::setTags(const QString& tags)
 {
-    set(m_data.tags, tags);
+    static QRegExp rx("(\\,|\\t|\\;)");
+    auto taglist = tags.split(rx, QString::SkipEmptyParts);
+    std::sort(taglist.begin(), taglist.end());
+    set(m_data.tags, taglist);
+}
+
+void Entry::addTag(const QString& tag)
+{
+    auto taglist = m_data.tags;
+    if (!taglist.contains(tag)) {
+        taglist.append(tag);
+        std::sort(taglist.begin(), taglist.end());
+        set(m_data.tags, taglist);
+    }
+}
```

Comment:
I think we should strip leading and trailing white space from (new) tag names.

Reply:
Yes good point we should do some more data cleansing in this function.